lib/puppet/network/http/response.rb

```diff
@@ -5,7 +5,23 @@ def initialize(handler, response)
   end
 
   def respond_with(code, type, body)
-    @handler.set_content_type(@response, type)
+    format = Puppet::Network::FormatHandler.format_for(type)
+    mime = format.mime
+    charset = format.charset
+
+    if charset && body.is_a?(String)
+      # REMIND: not all charsets are valid ruby encodings, e.g. ISO-2022-KR
+      encoding = Encoding.find(charset)
```

Comment:
Should raise an appropriate HTTP error if the charset is unknown. In practice shouldn't happen since it's either `utf-8` or `nil`, but that could change in the future.